src/internal/connector/data_collections_test.go

```diff
@@ -302,6 +344,61 @@ func (suite *ConnectorCreateExchangeCollectionIntegrationSuite) TestMailFetch()
 	}
 }
 
+func (suite *ConnectorCreateExchangeCollectionIntegrationSuite) TestDelta() {
+	ctx, flush := tester.NewContext()
+	defer flush()
+
+	var (
+		userID = tester.M365UserID(suite.T())
+		gc     = loadConnector(ctx, suite.T(), Users)
+	)
+
+	tests := []struct {
+		name  string
+		scope selectors.ExchangeScope
+	}{
+		{
+			name: "Mail",
+			scope: selectors.NewExchangeBackup().MailFolders(
+				[]string{userID},
+				[]string{exchange.DefaultMailFolder},
+				selectors.PrefixMatch(),
+			)[0],
+		},
+	}
+	for _, test := range tests {
+		suite.T().Run(test.name, func(t *testing.T) {
+			collections, err := gc.createExchangeCollections(ctx, test.scope, nil)
+			require.NoError(t, err)
+			assert.Less(t, 1, len(collections), "found metadata and data collections")
+
+			var metadata data.Collection
+
+			for _, coll := range collections {
+				if coll.FullPath().Service() == path.ExchangeMetadataService {
+					metadata = coll
+				}
+			}
+
+			require.NotNil(t, metadata, "found metadata collection")
+
+			_, deltas, err := parseMetadataCollections(ctx, []data.Collection{metadata})
+			require.NoError(t, err)
+
+			collections, err = gc.createExchangeCollections(ctx, test.scope, deltas)
+			require.NoError(t, err)
+
+			for _, coll := range collections {
+				if coll.FullPath().Service() != path.ExchangeMetadataService {
+					ec, ok := coll.(*exchange.Collection)
```

Comment:
is there a reason to specifically check this? From the perspective of all of the stuff outside of the `exchange` package the only thing that matters is that we get a `data.Collection` for everything

Reply:
What I actually wanted to check is that "the query that uses a delta token should be different/empty relative to the previous query which didn't have a delta token".  It's not a very functional test at the moment (I added a TODO), but I underlying concept is useful.

My brain was burned out friday night and I had trouble coming up with a proper set of criteria.  I'll probably still punt that to a different PR, but wanted to establish a place for the test to mark that spot.

Reply:
I'd prefer a TODO containing this info instead of some test check that's only vaguely related. Code has a tendency to stick around longer than we expect sometimes, so it's often better to be a bit more conservative with stuff like this

Reply:
Understood.  I think there's still remaining value, even in a just having it as a sanity test.  Not a strongly held opinion, in case you feel strongly about the todo.  I'm also feeling a little lax because I intend to fulfill the TODO her in a day or two, as soon as we've passed the previousPath and item comparison.  Revisiting the tests will be necessary to verify the changes as we tie all the work together..